test/manipulation.test.js

```diff
@@ -2662,6 +2662,25 @@ describe('manipulation', function() {
         null, // databases representing `undefined` as `null` (e.g. SQL)
       ]);
     });
+
+    it('Preserve custom type of auto-generated id property', async () => {
+      const userId = 'userId';
+
+      const User = db.define('User', {
+        id: {
+          type: String,
+          id: true,
+          useDefaultIdType: false,
+        },
+        name: String,
+      });
+
```

Comment:
As you have correctly mentioned, this test is already passing, even without your changes in place.

Here is the reason - the id type is changed only when the property is configured as `generated: true`. Only in that case the `useDefaultIdType` flag is changing the behavior. The flag is ignored for non-generated primary key, which is the case of your test.

Reply:
true, my fault